pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -0,0 +1,347 @@
+	if !ok1 || !ok2 {
+		return
+	}
+	oldPlantJSON, _ := json.Marshal(oldPlant)
+	newPlantJSON, _ := json.Marshal(newPlant)
+	plantLogger := logger.NewShootLogger(logger.Logger, newPlant.ObjectMeta.Name, newPlant.ObjectMeta.Namespace, "")
+
+	plantLogger.Debugf(string(oldPlantJSON))
+	plantLogger.Debugf(string(newPlantJSON))
+
+	c.plantAdd(newObj)
+}
+
+func (c *Controller) plantDelete(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+	c.plantQueue.Add(key)
+}
+
+func (c *Controller) reconcilePlantKey(ctx context.Context, key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	plant, err := c.plantLister.Plants(namespace).Get(name)
+	if apierrors.IsNotFound(err) {
+		logger.Logger.Debugf("[PLANT RECONCILE] %s - skipping because Plant has been deleted", key)
+		return nil
+	}
+	if err != nil {
+		logger.Logger.Infof("[PLANT RECONCILE] %s - unable to retrieve object from store: %v", key, err)
+		return err
+	}
+	if err := c.plantControl.Reconcile(ctx, plant, key); err != nil {
+		return err
+	}
+
+	c.plantQueue.AddAfter(key, c.config.Controllers.Plant.SyncPeriod.Duration)
+
+	return nil
+
+}
+
+// ControlInterface implements the control logic for updating Plants. It is implemented as an interface to allow
+// for extensions that provide different semantics. Currently, there is only one implementation.
+type ControlInterface interface {
+	Reconcile(context.Context, *gardencorev1alpha1.Plant, string) error
+}
+
+// NewDefaultPlantControl returns a new instance of the default implementation ControlInterface that
+// implements the documented semantics for Plants. updater is the UpdaterInterface used
+// to update the status of Plants.
+func NewDefaultPlantControl(k8sGardenClient kubernetes.Interface, recorder record.EventRecorder, config *config.ControllerManagerConfiguration, plantsLister gardencorelisters.PlantLister, secretLister kubecorev1listers.SecretLister) ControlInterface {
+	return &defaultPlantControl{
+		k8sGardenClient: k8sGardenClient,
+		plantClient:     make(map[string]client.Client),
+		discoveryClient: make(map[string]*kubernetesclientset.Clientset),
+		healthChecker:   make(map[string]*HealthChecker),
+		plantLister:     plantsLister,
+		secretsLister:   secretLister,
+		recorder:        recorder,
+		config:          config,
+	}
+}
+
+func (c *defaultPlantControl) Reconcile(ctx context.Context, obj *gardencorev1alpha1.Plant, key string) error {
+	var (
+		plant  = obj.DeepCopy()
+		logger = logger.NewFieldLogger(logger.Logger, "plant", plant.Name)
+	)
+
+	if plant.DeletionTimestamp != nil {
+		return c.delete(plant, logger)
+	}
+
+	return c.reconcile(ctx, plant, key, logger)
+}
+
+func (c *defaultPlantControl) conditionThresholdsToProgressingMapping() map[gardencorev1alpha1.ConditionType]time.Duration {
```

Comment:
I think we don't need the progressing status in the conditions here, right? We don't have any control, so let's just show the reality always.
@vlerenc thoughts?

(You are anyway taking configuration from other controllers which would require refactoring if we would want to keep this)